Source/TK.CustomMap/MapSpan.cs

```diff
@@ -17,6 +17,14 @@ public MapSpan(Position center, double latitudeDegrees, double longitudeDegrees)
             LongitudeDegrees = Math.Min(Math.Max(longitudeDegrees, MinimumRangeDegrees), 180.0);
         }
 
+        public MapSpan(Position ne, Position sw)
```

Comment:
this might be best as a separate proposal. I've thought about having this same thing before and I think there are potential pitfalls with it.

Reply:
Why native code Region/Span been limit by some point but in TK Region/Span been limit by Center and Radius. How difference with Region and Map Center + Zoom Level ?

Reply:
the issue is that it's really easy to pass in parameters that won't behave as you would want.

1) how does it behave if the distance between N + S is 200 Meters and the distance between E + W is 200 Kilometers?

2) how does it behave if someone passes in nw + se by accident?

Essentially if you pick a single point and then a radius, it'll be far less prone to error.

Reply:
- NS EW or Radius Center ... i think they are not different, they are in a rectangle.
![untitled](https://user-images.githubusercontent.com/25890459/44185940-ab606400-a140-11e8-858d-0ac5bcf4ed5a.jpg)

Reply:
plug this in and tell me what happens

```
// this IS a valid rectangle that fits the scenario above
var ne = new Position {
    Latitude = 47.34,
    Longitude = -88.97
};
var sw = new Position {
    Latitude = 47.33,
    Longitude = -121.31
};
```

```
// this is an INVALID rectangle based on the scenario above
var ne = new Position {
    Latitude = 47.33,
    Longitude = -88.97
};
var sw = new Position {
    Latitude = 47.34,
    Longitude = -121.31
```

Reply:
Your example like Center and Radius = 0 or radius so tiny

Reply:
yep, but as a user I might expect it to round up to the radius of East/West. The point is that the behavior is subject interpretation. With Center/Radius there's no subjectivity.